homework_4

```diff
@@ -0,0 +1,41 @@
+4. run tcpdump and request any http site in separate session. 
+Find HTTP request and answer packets with ASCII data in it.  
+Tcpdump command must be as strict as possible to capture only needed packages for this http request.
+
+(in the first session)
+[student@localhost chapter9]$ curl google.com
+
+(meanwhile in the second session)
+[student@localhost chapter9]$ sudo tcpdump tcp port 80 -A
```

Comment:
1. What else do we need to specify to get packets related exactly to our connection with `google.com` ?
2. Are you sure that port 80 was not reconfigured to another service/protocol on the destination side?

Reply:
1. Probably the next command will help us to specify connection:
[student@localhost ~]$ sudo tcpdump -A host google.com
2. I'm not sure about this because every host can reconfigure every port for different services. As default the destination web-host could work on port 443 (for HTTPS) and port 80 (for HTTP). So probably it was just my assumption about port 80

Reply:
1. In that case you better add to definitions at least the interface and protocol
2. Since we're catching the ASCII data, we can filter our packets by its' content, for example, we can add an ASCII filter to catch only HTTP requests.

Reply:
Did you mean that we can filter http requests like this?
[student@localhost ~]$ sudo tcpdump -i enp0s3 tcp port http -A

Reply:
This one too, but we can add ASCII filter which is looks like 'tcp[((tcp[12:1] & 0xf0) >> 2):4] = 0x48545450'